### PR TITLE
HAL legacy support

### PR DESCRIPTION
**Pull request by fpistm**
This PR introduce support of HAL Legacy drivers and enable HAL Ethernet legacy support for the Nucleo F429ZI.

Note: Several scripts have been updated to reach this and some fixes have been done during this PR development.

### CODE REVIEW

File: variants/STM32F0xx/F042C(4-6)(T-U)/PeripheralPins.c [hunk 1]
```diff
@@ -199,15 +199,15 @@ WEAK const PinMap PinMap_SPI_SSEL[] = {
 
 //*** CAN ***
 
-#ifdef HAL_CAN_MODULE_ENABLED
+#if defined(HAL_CAN_MODULE_ENABLED) || defined(HAL_CAN_LEGACY_MODULE_ENABLED)
```
Comment (ABOSTM) [inline variants/STM32F0xx/F042C(4-6)(T-U)/PeripheralPins.c:202]: All generic PeripheralPins.c have been changed. Great.
Is it necessary to apply the same change in custom PeripheralPins.c like for example PeripheralPins_BLACK_F407VX.c ?

**Review by ABOSTM — CHANGES_REQUESTED** [review]
Typo in the 12th commit: 
"variant(G4): update properly **against** open pin data DB 6.0.50"

File: CI/update/stm32variant.py [hunk 2]
```diff
@@ -816,7 +816,11 @@ def can_pinmap(lst):
         )
     return dict(
         name=name,
-        hal=name,
+        hal=name
+        if name == "FDCAN"
+        else ["CAN", "CAN_LEGACY"]
+        if any(mcu in mcu_family for mcu in legacy_hal["CAN"])
+        else name,
```
[the remark this answers is not in the document]
Comment (fpistm) [reply CI/update/stm32variant.py:821]: Done

**Review by ABOSTM — APPROVED** [review]
LGTM

**Pull request merged**